bin/reth/src/commands/common.rs

```diff
@@ -0,0 +1,115 @@
+//! Contains common `reth` arguments
+
+use clap::Parser;
+use reth_config::{config::EtlConfig, Config};
+use reth_db::{init_db, open_db_read_only, DatabaseEnv};
+use reth_db_common::init::init_genesis;
+use reth_node_core::{
+    args::{
+        utils::{chain_help, genesis_value_parser, SUPPORTED_CHAINS},
+        DatabaseArgs, DatadirArgs,
+    },
+    dirs::{ChainPath, DataDirPath},
+};
+use reth_primitives::ChainSpec;
+use reth_provider::{providers::StaticFileProvider, ProviderFactory};
+use std::{path::PathBuf, sync::Arc};
+use tracing::{debug, info};
+
+/// Struct to hold config and datadir paths
+#[derive(Debug, Parser)]
+pub struct EnvironmentArgs {
+    /// Parameters for datadir configuration
+    #[command(flatten)]
+    pub datadir: DatadirArgs,
+
+    /// The path to the configuration file to use.
+    #[arg(long, value_name = "FILE", verbatim_doc_comment)]
+    pub config: Option<PathBuf>,
+
+    /// The chain this node is running.
+    ///
+    /// Possible values are either a built-in chain or the path to a chain specification file.
+    #[arg(
+        long,
+        value_name = "CHAIN_OR_PATH",
+        long_help = chain_help(),
+        default_value = SUPPORTED_CHAINS[0],
+        value_parser = genesis_value_parser
+    )]
+    pub chain: Arc<ChainSpec>,
+
+    /// All database related arguments
+    #[command(flatten)]
+    pub db: DatabaseArgs,
+}
+
+impl EnvironmentArgs {
+    /// Initializes environment according to [`AccessRights`] and returns a tuple of [Config] &
+    /// [`ProviderFactory`].
```

Comment:
it's not a tuple anymore
```suggestion
    /// Initializes environment according to [`AccessRights`].
```